src/shogun/base/SGObject.h

```diff
@@ -996,6 +989,21 @@ class CSGObject
 	void register_observable(
 		const std::string& name, const std::string& description);
 
+	/**
+	 * Get the current step for the observed values.
+	 */
+#ifndef SWIG
+	SG_FORCED_INLINE int64_t get_step()
+	{
+		int64_t step = -1;
+		Tag<int64_t> tag("current_iteration");
```

Comment:
this way of doing things will tie and observed value to be 'stepped' by `current_iteration` object tag... which should not be the case **always**. could be but not necessary

Reply:
This was preliminary.
Let’s make it virtual then!?

Reply:
noup... it's the wrong place imo. discussed with @geektoni he'll try to address it later

Reply:
What’s the solution here then?